src/nupic/math/ArrayAlgo.hpp

```diff
@@ -3779,7 +3779,7 @@ namespace nupic {
       // n1 is the number floats we can process in parallel with xmm
       // n2 is the number of "stragglers" what we will have to do one by one ( < 4)
       nupic::Real32 count = 0;
-      long x_addr = (long) begin; // 8 bytes on 64 bits platforms
+      long long x_addr = (long long) begin; // 8 bytes on 64 bits platforms
```

Comment:
Alternatives could be `size_t` or `uintptr_t`? Both cover LLP64 and LP64 data models (http://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models). Ideally a new NTA basic type for pointer manipulation :)